server/cmwell-bg/src/main/scala/cmwell/bg/ImpStream.scala

```diff
@@ -820,16 +837,16 @@ class ImpStream(partition: Int, config: Config, irwService: IRWService, zStore:
       val manageIndices = builder.add(
         Flow[BGMessage[(BulkIndexResult, Seq[IndexCommand])]].map {
           case bgMessage@BGMessage(_, (bulkRes, commands)) =>
-            indexCount += bulkRes.successful.size
-            if (indexCount / numOfShardPerIndex >= maxDocsPerShard) {
+            startingIndexCount += bulkRes.successful.size
+            if (startingIndexCount / numOfShardPerIndex >= maxDocsPerShard) {
               val (pref, suf) = currentIndexName.splitAt(currentIndexName.lastIndexOf('_') + 1)
               val nextCount = suf.toInt + 1
               val nextIndexName = pref + nextCount
               ftsService.createIndex(nextIndexName).flatMap { _ =>
                 scheduleFuture(10.seconds)(ftsService.updateAllAlias)
               }.andThen { case _ =>
                 currentIndexName = nextIndexName
-                indexCount = 0
+                startingIndexCount = 0
```

Comment:
this line run in the context of the fts Future. The read is Flow of the stream. Those are different threads!
1. The Flow thread might not see the value (on solaris it's the case)
2. even on our architecture (CPU, Linux) it might see the update only later. Think about it: the `startingIndexCount = 0` will be at least 10 seconds after the initial "decision" to create a new index. During that time ingest keeps running and `startingIndexCount / numOfShardPerIndex >= maxDocsPerShard` condition will be met and a new index will be created.